### DISCUSSION TITLE
Convert not working in Command line but not in java process runtime

### DISCUSSION THREAD

**Discussion by cmac45** [ImageMagick/ImageMagick#5031] [General]
Hi, 

I have a tif image where I am trying to draw a box and compress the image with LZW compression simultaneously. 

this is the command I am running, and it works fine from windows command line. 
`C:\ImageMagick-7.1.0\convert.exe "C:\Users\admin\Orig.tif" -draw "rectangle 576,1069,943,1114" -compress LZW "C:\Users\admin\DrawLZW.tif"`

when I try to execute the same command with my java program, I get an image file created, but the file size is 1kb 

  Process p = Runtime.getRuntime().exec(commandPNG);
            p.waitFor();

            String[] cmd = {"C:\\ImageMagick-7.1.0\\convert.exe", "\"C:\\Users\\chris.macwilliams\\Orig.tif\"", "-draw", "\"rectangle 576,1069,943,1114\"", "–compress","LZW", "\"C:\\Users\\chris.macwilliams\\DrawLZWwithJava.tif\""};
            LOGGER.info(cmd);
            Process pt = Runtime.getRuntime().exec(cmd);
            pt.waitFor();

            if (pt.exitValue() != 0) {
                LOGGER.error("ERROR with Image Magic Command exit value:" + pt.exitValue()+  " "+ commandTIF);

Any Ideas here? 

Using IM Version: ImageMagick-7.1.0
I have included the test images that I am getting the errors on. 
[images.zip](https://github.com/ImageMagick/ImageMagick/files/8454134/images.zip)

**Comment by snibgo** [comment]
I know nothing about Java.

For IM v7, I suggest using "magick" instead of "convert".

As an output file is created, we know "convert" has been run. But the output file is corrupt. This suggests the process was killed externally. Perhaps IM requested a resource that Java wouldn't or couldn't provide.

I suggest experimenting with Java. Can it execute system commands, such as copying one file to another? Does the IM command work with small input images?

**Reply by cmac45** [reply]
So I just did test magick.exe vs  convert.exe, and I got the result that I was expecting. For our production deployment we don't include the magick.exe in the build so I will have to figure that out. I would have thought convert would have performed the same way.

**Comment by burnellm** [comment]
I've added the solution here: https://stackoverflow.com/a/71946010/4654863
